### PR TITLE
Add API endpoint documentation for doubts and doubtAnswers

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
Add API endpoint documentation for doubts and doubtAnswers

1). adds APIPARAMS of courseid, doubtId, doubtAnswerId
2). add responsedoc
3). filled the example values for both doubts and doubt answer in for the swagger documnetation

**Type of Change:**

- [x] Code
- [x] New Feature
- [x] Documentation

**How Has This Been Tested?**

Terminal commands to test code:-
1). npm run lint
2). npm run lint:fix
3). nest start
4). npm run test

**Checklist**

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Review by Devesh21700Kumar — COMMENTED** [review]
@Abhishek-kumar09 the doubts and doubtAnswers API documentation here

File: src/modules/course/course.service.ts [hunk 1]
```diff
@@ -31,6 +36,14 @@ export class CourseService {
...
         .populate('schedule')
         .populate('reviews')
+        .populate({
+          path: 'doubts',
+          model: 'Doubt',
+          populate: {
+            path: 'answers',
+            model: 'DoubtAnswer',
+          },
+        })
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/modules/course/course.service.ts:46]: So I should make a seperate endpoint for getting doubts for course right?

[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/modules/course/course.service.ts:46]: @Abhishek-kumar09 
https://github.com/codeforcauseorg/edu-server/pull/108/commits/cc6cee520cd2b37e1696e283035609129dc3a374

This commit has the required changes made
Please check PR 108
This PR is just a continuation for the documentation

File: src/modules/course/course.module.ts [hunk 2]
```diff
@@ -5,13 +5,17 @@ import { CourseController } from './course.controller';
 import { CourseService } from './course.service';
 import { ScheduleSchema } from './schema/schedule.schema';
 import { ReviewSchema } from './schema/review.schema';
+import { DoubtSchema } from 'modules/doubt/schema/doubt.schema';
+import { DoubtAnswerSchema } from 'modules/doubt/schema/doubtAnswer.schema';
```
Comment (Abhishek-kumar09) [inline src/modules/course/course.module.ts:9]: put relative path, these aboslute paths are creating problems in build

File: src/modules/course/course.service.ts [hunk 3]
```diff
@@ -16,13 +16,18 @@ import { UpdateScheduleDto } from './dto/update-schedule.dto';
 import { CreateReviewDto } from './dto/create-review.dto';
 import { UpdateReviewDto } from './dto/update-review.dto';
 import { Review } from './schema/review.schema';
+import { Doubt } from 'modules/doubt/schema/doubt.schema';
+import { DoubtAnswer } from 'modules/doubt/schema/doubtAnswer.schema';
```
Comment (Abhishek-kumar09) [inline src/modules/course/course.service.ts:20]: Here too

File: src/modules/doubt/schema/doubtAnswer.schema.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import * as mongoose from 'mongoose';
```
Comment (Abhishek-kumar09) [inline src/modules/doubt/schema/doubtAnswer.schema.ts:2]: Document

Comment (Devesh21700Kumar) [reply src/modules/doubt/schema/doubtAnswer.schema.ts:2]: yeah i've fixed them in the original PR #108 
The present PR#109 is just a continuation for documentation, and all changes of PR#108 will be merged in that later by me

**Pull request closed without merging**